miniconf/src/node.rs

```diff
@@ -173,10 +173,47 @@ impl<T, const S: char> From<T> for Path<T, S> {
     }
 }
 
+/// String split/skip wrapper, smaller/simpler than `.split(S).skip(1)`
+#[derive(Copy, Clone, Default, Debug, PartialEq, PartialOrd, Serialize, Deserialize)]
+#[repr(transparent)]
+pub struct PathIter<'a, const S: char>(Option<&'a str>);
+
+impl<'a, const S: char> PathIter<'a, S> {
+    #[inline]
+    fn new(s: &'a str) -> Self {
+        let mut s = Self(Some(s));
+        // Skip the first part to disambiguate between
+        // the one-Key Keys `[""]` and the zero-Key Keys `[]`.
+        // This is relevant in the case of e.g. `Option` and newtypes.
+        // See the corresponding unittests (`just_option`).
+        // It implies that Paths start with the separator
+        // or are empty. Everything before the first separator is ignored.
+        s.next();
+        s
+    }
+}
+
+impl<'a, const S: char> Iterator for PathIter<'a, S> {
+    type Item = &'a str;
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.map(|s| {
+            let pos = s
+                .chars()
+                .map_while(|c| (c != S).then_some(c.len_utf8()))
+                .sum();
+            let (left, right) = s.split_at(pos);
+            self.0 = right.get(S.len_utf8()..);
```

Comment:
How is this semantically different than `split(S).skip(1)`?

Reply:
`PathIter` has exactly the same Iterator behavior as `Skip<Split<_>>`. It's just 2 kB less code on stabilizer.
https://github.com/quartiq/miniconf/blob/0cd1773c192d14ad5423595d7817fda9df161785/miniconf/src/node.rs#L306-L314

Reply:
Huh, interesting. Alright, seems fine to me.